src/gui/shareusergroupwidget.cpp

```diff
@@ -926,13 +937,54 @@ void ShareUserLine::customizeStyle()
     _deleteShareButton->setIcon(deleteicon);
 
     _ui->noteConfirmButton->setIcon(Theme::createColorAwareIcon(":/client/theme/confirm.svg"));
-    _ui->confirmExpirationDate->setIcon(Theme::createColorAwareIcon(":/client/theme/confirm.svg"));
     _ui->progressIndicator->setColor(QGuiApplication::palette().color(QPalette::WindowText));
 
     // make sure to force BackgroundRole to QPalette::WindowText for a lable, because it's parent always has a different role set that applies to children unless customized
     _ui->errorLabel->setBackgroundRole(QPalette::WindowText);
 }
 
+QPixmap ShareUserLine::pixmapForShareeType(Sharee::Type type, const QColor &backgroundColor) const
+{
+    QString fileName;
+    switch (type) {
+    case Sharee::Room:
+        fileName = QStringLiteral("talk-app.svg");
+        break;
+    case Sharee::Email:
+        fileName = QStringLiteral("email.svg");
+        break;
+    default:
+        return QPixmap();
+    }
+
+    return Theme::instance()->pixmapForBackground(fileName, backgroundColor);
+}
+
+QColor ShareUserLine::backgroundColorForShareeType(Sharee::Type type) const
+{
+    switch (type) {
+    case Sharee::Room:
+        return Theme::instance()->wizardHeaderBackgroundColor();
+    case Sharee::Email:
+        return Theme::instance()->wizardHeaderTitleColor();
+    default:
+        break;
+    }
+
+    const auto calculateBackgroundBasedOnText = [this]() {
```

Comment:
Not sure why it is not simply a static function inside the `cpp` file.

Reply:
@FlexW In this case, it makes little sense because we only use it in one place inside this `cpp`.

Reply:
I think it does make sense even if it is just one function. The function `backgroundColorForShareeType()` will become smaller and you directly see what it does :)

Reply:
@FlexW But why isn't it obvious what it does now? It is implied by its name? Or not? The only thing that's been added is the switch-case at the beginning and it is still returning a `QColor` all the time. I am still in doubt why we should create tons of static functions. As a compromise, I can create a function in an anonymous namespace, if you think that would help improve readability.

Reply:
Go with whatever you like. That was just a suggestion. I just think that lambdas harm readability because it clutters the outer function (in this case `ShareUserLine::backgroundColorForShareeType()`). A static function would not be defined inside the function and because of that you would faster see what `ShareUserLine::backgroundColorForShareeType()` does.

Reply:
WIth clutter I mean that in the function `ShareUserLine::backgroundColorForShareeType()` now a lot of other stuff happens that are unrelated to the function and in that the reader is not interested at this point.